app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt

```diff
@@ -268,7 +269,11 @@ class NavigationOverlayFragment : Fragment() {
                 FxaRepo.AccountState.AUTHENTICATED_WITH_PROFILE -> {
                     fxaRepo.accountManager.accountProfile()?.let { profile ->
                         if (profile.avatar != null) {
```

Comment:
You should check `url` is non-null here instead of below. In the current code, if url is null but avatar is not, we're not going to update the icon properly.